utils/visualizer/qviz/content_loader.py

```diff
@@ -2,250 +2,264 @@
+        df = delta_table.get_add_actions(True).to_pandas()
+        # Convert all blocks JSON chains to a list of dictionaries
+        df['tags.blocks'] = df['tags.blocks'].apply(lambda x: json.loads(x) if isinstance(x, str) else x)
+        # Make sure that 'tags.blocks' is a list of dictionaries
+        exploded_df = df.explode('tags.blocks')
+        # Extract 'cubeId'
+        all_cube_ids = exploded_df['tags.blocks'].apply(lambda x: x['cubeId'] if isinstance(x, dict) and 'cubeId' in x else None).dropna().tolist()
+
+        for cube_id in all_cube_ids:
+            cube_encoding_size = len(cube_id)
+            if 0 < cube_encoding_size < symbol_count:
+                symbol_count = cube_encoding_size
+        return None, symbol_count
+
+
+# 3. Returns a dictionary, the keys are the cubeids and the values are the actual cubes
+def extract_cubes_from_blocks(delta_table: DeltaTable , symbol_count: int) -> dict:
     """
-    Extract valid AddFiles from a single json log file. An AddFile is valid
-    if there's no RemoveFile with the same path.
-    example log file entries:
-    {"metaData":{"id":"c109cfb6-a770-4c02-b004-d44bbe91e981", ...}}
-    {"add":{"path":"b4542891-5b03-40cc-8ef1-293493e21814.parquet", ...}}
-    {"remove":{"path":"272bbd79-2bf0-444d-b15e-8e4326c9d281.parquet", ...}}
-    :param file_path: json log file path
-    :param revision_id: target RevisionID
-    :return: a list of AddFiles, and a set of RemoveFile paths
+    Takes all the blocks stored in the table and creates the cubes.
+    :param delta_table: list of json log files
+    :param symbol_count: Integer representing the number of symbols used for the representation
+    :return: A dictionary with all the cubes stored in the table.
     """
-    with open(file_path, 'r') as f:
-        lines = f.readlines()
-
-    add_files = list()
-    remove_paths = set()
-    for line in lines:
-        action_file = json.loads(line)
-        # AddFile entry
-        if 'add' in action_file:
-            add_file = action_file['add']
-            _id = add_file['tags']['revision']
-            if _id == revision_id:
-                # Add revision AddFile to result
-                add_files.append(add_file)
-        # RemoveFile entry
-        elif 'remove' in action_file:
-            remove_file = action_file['remove']
-            remove_paths.add(remove_file['path'])
-
-    return add_files, remove_paths
-
-
-def extract_revision_metadata(checkpoint_file: str, json_files: list[str], revision_id: str) -> dict:
+
+    df = delta_table.get_add_actions(True).to_pandas()
+
+    df_filtered = df[['size_bytes', 'tags.blocks']]
+    # In this dictionary we store a cube for every cubeId. Each cube will be associated to his cube ID
+    cubes_dict = dict()
+
+    # Iterate throughout each row of the dataframe
+    for index, row in df_filtered.iterrows():
+        cube = row["tags.blocks"] # This returns a pandas series with the blocks of the cube in each of the iterated rows
```

Comment:
Make sure you are naming these variables correctly.
Here, the list of blocks should be called `blocks` instead.